### PR TITLE
Prediction: initial implement of prediction scenario feature

### PR DESCRIPTION
**Pull request by kechxu**

### CODE REVIEW

File: modules/prediction/scenario/feature_extractor/feature_extractor.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  void ExtractNeighborLaneFeatures(
+      const std::shared_ptr<const hdmap::LaneInfo>& ptr_ego_lane,
+      const common::math::Vec2d& ego_position);
+
+  void ExtractFrontJunctionFeatures();
+
+  void ExtractObstacleFeatures();
+
+  std::shared_ptr<const hdmap::LaneInfo> GetEgoLane(
```
Comment (SohamGhormade) [inline modules/prediction/scenario/feature_extractor/feature_extractor.h]: Please clarify what Ego refers to.Thanks in advance

File: modules/prediction/container/pose/pose_container.h [hunk 2]
```diff
@@ -54,13 +54,31 @@ class PoseContainer : public Container {
...
-  apollo::perception::PerceptionObstacle* ToPerceptionObstacle();
+  perception::PerceptionObstacle* ToPerceptionObstacle();
 
   /**
    * @brief Get timestamp
    */
   double GetTimestamp();
 
+  /**
+   * @brief Get speed of adc
```
Comment (SohamGhormade) [inline modules/prediction/container/pose/pose_container.h]: minor ,abbreviation could be made upper case.
i.e ADC for autonomous driving car.

File: modules/prediction/scenario/feature_extractor/feature_extractor.h (hunk 1 above)
Comment (YajiaZhang) [reply modules/prediction/scenario/feature_extractor/feature_extractor.h]: Ego vehicle

File: modules/prediction/scenario/scenario_manager.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+ private:
+  FeatureExtractor feature_extractor_;
+  ScenarioAnalyzer scenario_analyzer_;
+};
+
+}  // namespace prediction
+}  // namespace apollo
+
+#endif  // MODULES_PREDICTION_SCENARIO_SCENARIO_MANAGER_H_
```
[the remark this answers is not in the document]
Comment (kechxu) [reply modules/prediction/scenario/scenario_manager.h]: I have added the unit tests.

File: modules/prediction/container/adc_trajectory/adc_trajectory_container.h [hunk 4]
```diff
@@ -80,7 +81,25 @@ class ADCTrajectoryContainer : public Container {
...
    * @brief Set ADC position
    */
-  void SetPosition(const ::apollo::common::math::Vec2d& position);
+  void SetPosition(const common::math::Vec2d& position);
+
+  /**
+   * @brief Get ADC junction
+   * @return A pointer to ADC junction information
+   */
+  std::shared_ptr<const hdmap::JunctionInfo> ADCJunction() const;
```
[the remark this answers is not in the document]
Comment (kechxu) [reply modules/prediction/container/adc_trajectory/adc_trajectory_container.h]: I have added the unit test

File: modules/prediction/container/pose/pose_container.h (hunk 2 above)
Comment (YajiaZhang) [reply modules/prediction/container/pose/pose_container.h]: Thanks for the comments. It will be updated in next version.

**Pull request merged**